### PR TITLE
Code Analysis: add FAQ

### PR DESCRIPTION
**Pull request by juli1**
### What does this PR do? What is the motivation?

We add a list of frequently asked questions that customers asked us during calls. This way, we can easily refer them to the list of questions and customers can proactively learn about the capabilities of our tools.

### CODE REVIEW

File: content/en/code_analysis/faq/_index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  text: "Learn about Code Analysis"
+- link: "/code_analysis/static_analysis/"
+  tag: "Documentation"
+  text: "Learn about Static Analysis"
+- link: "/code_analysis/software_composition_analysis/"
+  tag: "Documentation"
+  text: "Learn about Software Composition Analysis"
+---
+
+## Common Questions
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:16]:
```suggestion
## Common questions
```

File: content/en/code_analysis/faq/_index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+### What are Datadog-hosted scans?
+
+When using Datadog-hosted scan, Datadog scans your code within Datadog infrastructure. Datadog acts
+as a CI pipeline: it clones your code, runs the analyzer and uploads the results for you.
+
+With Datadog-hosted scans, you do not need to do the heavy lifting and configure any CI pipeline to run the analyzer.
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:23]:
```suggestion
When using a Datadog-hosted scan, Datadog scans your code within the Datadog infrastructure. Datadog acts
as a CI pipeline: it clones your code, runs the analyzer, and uploads the results for you.

With Datadog-hosted scans, you do not need to do the heavy lifting and configure any CI pipeline to run the analyzer.
```

File: content/en/code_analysis/faq/_index.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+## Static Analysis
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:25]:
```suggestion
## Static analysis
```

File: content/en/code_analysis/faq/_index.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+### May I import results from other analyzers into Datadog?
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:27]:
```suggestion
### Can results be imported into Datadog from other analyzers?
```

File: content/en/code_analysis/faq/_index.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+While we recommend to use the Datadog Static Analyzer tool[4], Datadog supports the ingestion of files that enforces the SARIF format.
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:29]:
```suggestion
While the [Datadog Static Analyzer tool][4] is recommended, Datadog supports the ingestion of files that enforce the SARIF format.
```

File: content/en/code_analysis/faq/_index.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+At this time, we tested our API with the following tools:
+
+ - [gitleaks][1]
+ - [semgrep][2]
+ - [eslint][3]
+
+To ingest your SARIF file into Datadog:
+ - Install the `datadog-ci` cli (requires nodejs to be installed)
+ - Ensure that your `DD_SITE`, `DD_API_KEY` and `DD_APP_KEY` environment variables are set
+ - Invoke the tool to upload the file to Datadog.
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:40]:
```suggestion
1. Install the `datadog-ci` CLI (requires that Node.js is installed).
2. Ensure that your `DD_SITE`, `DD_API_KEY`, and `DD_APP_KEY` environment variables are set.
3. Invoke the tool to upload the file to Datadog.
```

File: content/en/code_analysis/faq/_index.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+```bash
+# Install datadog-ci
+npm install -g @datadog/datadog-ci
+
+# Upload SARIF file
+datadog-ci sarif upload --service "your-app" --env "ci" /path/to/sarif-file.json
+```
+
+If you want to import from a tool not support at this time, please contact your Customer Success Manager.
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:52]:
```suggestion
If you want to import using a tool that is not supported currently, contact your Customer Success Manager.
```

File: content/en/code_analysis/faq/_index.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+### Can I write custom rules?
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:54]:
```suggestion
### Can custom rules be used?
```

File: content/en/code_analysis/faq/_index.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+Custom rules will be available to all beta customers at the end of Q3 2024.
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:56]:
```suggestion
Custom rule availability is planned for all beta customers at the end of Q3 2024.
```

File: content/en/code_analysis/faq/_index.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+scans the files that changed between the current branch and the default branch. Diff-aware scans
+last seconds while full-scans may take a few minutes (depending on the codebase).
+
+In order for diff-aware to scan, you need to be using the Datadog platform.
+
+### What is Datadog Static Analyzer on the OWASP benchmark?
+
+Our Static Analyzer has been tested against the OWASP benchmark with a score of 44.
+We periodically update the results against the benchmark and publish the results in
+the [static analyzer documentation][5]
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:72]:
```suggestion
Our Static Analyzer has been tested against the OWASP benchmark with a score of 44. We periodically update the results against the benchmark and publish the results in the [static analyzer documentation][5].
```

File: content/en/code_analysis/faq/_index.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+### Is the analyzer open-source?
+
+The Datadog Static Analyzer is available under an open-source license and the code
+is [available on GitHub][4].
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:77]:
```suggestion
The Datadog Static Analyzer is available under an open-source license and the code is [available on Github][4].
```

File: content/en/code_analysis/faq/_index.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+## Software Composition Analysis
+
+### May I import any SBOM file in Datadog?
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:81]:
```suggestion
### Can I import any SBOM file into Datadog?
```

File: content/en/code_analysis/faq/_index.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+
+While we recommend to use Datadog SBOM generator tool[6], Datadog supports the ingestion of any SBOM files.
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:83]:
```suggestion
While the [Datadog SBOM generator tool][6] is recommended, Datadog supports the ingestion of any SBOM files.
```

File: content/en/code_analysis/faq/_index.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+Datadog only supports SBOM file with the Cyclone-DX 1.4 and Cyclone-DX 1.5 formats.
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:84]:
```suggestion
Datadog only supports SBOM files with the Cyclone-DX 1.4 and Cyclone-DX 1.5 formats.
```

File: content/en/code_analysis/faq/_index.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+
+At this time, we tested our CLI tools with the following tools:
+
+- [osv-scanner][6]
+- [trivy][7]
+
+To ingest your SARIF file into Datadog:
+- Install the `datadog-ci` cli (requires nodejs to be installed)
+- Ensure that your `DD_SITE`, `DD_API_KEY` and `DD_APP_KEY` environment variables are set
+- Invoke the tool to upload the file to Datadog.
```
Comment (michaelcretzman) [inline content/en/code_analysis/faq/_index.md:94]:
```suggestion
To ingest your SARIF file into Datadog:
1. Install the `datadog-ci` CLI (requires that Node.js is installed).
2. Ensure that your `DD_SITE`, `DD_API_KEY` and `DD_APP_KEY` environment variables are set.
3. Invoke the tool to upload the file to Datadog.
```

**Review by michaelcretzman — CHANGES_REQUESTED** [review]
Please incorporate the suggested edits.

**Review by michaelcretzman — APPROVED** [review]

File: content/en/code_analysis/faq/_index.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+- link: "/code_analysis/software_composition_analysis/"
+  tag: "Documentation"
+  text: "Learn about Software Composition Analysis"
+---
+
+## Common questions
+
+### What are Datadog-hosted scans?
+
+When using a Datadog-hosted scan, Datadog scans your code within the Datadog infrastructure. Datadog acts
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:20]: With Datadog-hosted scans, your code is scanned within Datadog's infrastructure as opposed to within your CI pipeline. Datadog clones your code, runs the static analyzer to perform Static Analysis and/or Software Composition Analysis, and uploads the results for you.

File: content/en/code_analysis/faq/_index.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+as a CI pipeline: it clones your code, runs the analyzer, and uploads the results for you.
+
+With Datadog-hosted scans, you do not need to do the heavy lifting and configure any CI pipeline to run the analyzer.
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:23]: The benefit of Datadog-hosted scans is that no configuration is needed in your CI pipeline(s) to use Code Analysis.

File: content/en/code_analysis/faq/_index.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+
+## Static analysis
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:25]: capitalize Analysis

File: content/en/code_analysis/faq/_index.md [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+
+### Can results be imported into Datadog from other analyzers?
+
+While the [Datadog Static Analyzer tool][4] is recommended, Datadog supports the ingestion of files that enforce the SARIF format.
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:29]: remove tool - "Datadog Static Analyzer"

Comment (kassenq) [reply content/en/code_analysis/faq/_index.md:29]: "files that adhere to SARIF format"

File: content/en/code_analysis/faq/_index.md [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+
+At this time, we tested our API with the following tools:
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:31]: Ingestion of SARIF files is verified for the following third-party tools:

File: content/en/code_analysis/faq/_index.md [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+
+If you want to import using a tool that is not supported currently, contact your Customer Success Manager.
+
+### Can custom rules be used?
+
+Custom rule availability is planned for all beta customers at the end of Q3 2024.
+
+### Do you always scan all the files at every push or commit?
+
+Scans on the default branches are always full scans.
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:60]: Remove the - "Scans on default branches are always full scans."

File: content/en/code_analysis/faq/_index.md [hunk 22]
```diff
@@ -0,0 +1,113 @@
...
+
+Scans on non-default branches use *diff-aware* scans. With diff-aware scans, the analyzer only
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:62]: By default, scans on non-default branches use ...

File: content/en/code_analysis/faq/_index.md [hunk 23]
```diff
@@ -0,0 +1,113 @@
...
+scans the files that changed between the current branch and the default branch. Diff-aware scans
+last seconds while full-scans may take a few minutes (depending on the codebase).
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:64]: last seconds, while

File: content/en/code_analysis/faq/_index.md [hunk 24]
```diff
@@ -0,0 +1,113 @@
...
+
+In order for diff-aware to scan, you need to be using the Datadog platform.
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:66]: You must be using Datadog's analyzer to enable diff-aware scans.

File: content/en/code_analysis/faq/_index.md [hunk 25]
```diff
@@ -0,0 +1,113 @@
...
+
+### What is Datadog Static Analyzer on the OWASP benchmark?
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:68]: "What is Datadog's OWASP benchmark score for Static Analysis?"

File: content/en/code_analysis/faq/_index.md [hunk 26]
```diff
@@ -0,0 +1,113 @@
...
+
+Our Static Analyzer has been tested against the OWASP benchmark with a score of 44. We periodically update the results against the benchmark and publish the results in the [static analyzer documentation][5].
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:70]: Datadog's Static Analyzer has been tested against the OWASP benchmark with a score of 44. The analyzer is periodically checked against the benchmark and updated results are published to the [static analyzer documentation][5].

File: content/en/code_analysis/faq/_index.md [hunk 27]
```diff
@@ -0,0 +1,113 @@
...
+
+### Is the analyzer open-source?
+
+The Datadog Static Analyzer is available under an open-source license and the code is [available on Github][4].
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:74]: can we specify the license?

File: content/en/code_analysis/faq/_index.md [hunk 28]
```diff
@@ -0,0 +1,113 @@
...
+
+## Software Composition Analysis
+
+### Can I import any SBOM file into Datadog?
+
+While the [Datadog SBOM generator tool][6] is recommended, Datadog supports the ingestion of any SBOM files.
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:80]: remove "tool"

File: content/en/code_analysis/faq/_index.md [hunk 29]
```diff
@@ -0,0 +1,113 @@
...
+Datadog only supports SBOM files with the Cyclone-DX 1.4 and Cyclone-DX 1.5 formats.
+
+At this time, we tested our CLI tools with the following tools:
```
Comment (kassenq) [inline content/en/code_analysis/faq/_index.md:83]: Ingestion of SBOM files is verified for the following third-party tools:

**Review by kassenq — CHANGES_REQUESTED** [review]

File: content/en/code_analysis/faq/_index.md [hunk 30]
```diff
@@ -0,0 +1,114 @@
...
+
+While the [Datadog SBOM generator][6] is recommended, Datadog supports the ingestion of any SBOM files.
+Datadog only supports SBOM files with the Cyclone-DX 1.4 and Cyclone-DX 1.5 formats.
+
+Ingestion of SBOM files is verified for the following third-party tools:
+
+- [osv-scanner][6]
+- [trivy][7]
+
+To ingest your SARIF file into Datadog:
```
Comment (dastrong) [inline content/en/code_analysis/faq/_index.md:89]:
```suggestion
To ingest your SBOM file into Datadog:
```

**Review by dastrong — APPROVED** [review]

**Pull request merged**